packages/react-ui/src/widgets/WrapperWidgetUI.js

```diff
@@ -30,79 +30,94 @@ Actions props must have this format:
 ];
 */
 
-const useStyles = makeStyles((theme) => ({
-  root: {
-    position: 'relative',
-    maxWidth: '100%',
-    padding: ({ margin }) => (margin !== undefined ? margin : theme.spacing(2, 2.5))
-  },
-  loading: {
-    position: 'absolute',
-    top: 0,
-    left: 0,
-    width: '100%',
-    height: theme.spacing(0.25)
+const Root = styled(Box)(({ theme, margin = false }) => ({
+  position: 'relative',
+  maxWidth: '100%',
+  padding: margin !== undefined ? margin : theme.spacing(2, 2.5)
+}));
+
+const LoadingBar = styled(LinearProgress)(({ theme }) => ({
+  position: 'absolute',
+  top: 0,
+  left: 0,
+  width: '100%',
+  height: theme.spacing(0.25)
+}));
+
+const Header = styled(Grid, {
+  shouldForwardProp: (prop) => prop !== 'expanded'
+})(({ theme, expanded = true }) => ({
+  display: 'flex',
+  flexDirection: 'row',
+  alignItems: 'flex-start',
+  justifyContent: 'space-between',
+  width: '100%',
+  ...(expanded ? { minHeight: theme.spacing(3) } : { height: theme.spacing(3) }),
+  padding: 0
+}));
+
+const HeaderButton = styled(Button)(({ theme, expandable = true }) => ({
+  flex: 1,
+  padding: 0,
+  alignItems: 'flex-start',
+  justifyContent: 'flex-start',
+  cursor: expandable ? 'pointer' : 'default',
+  '& .MuiButton-startIcon': {
+    marginTop: '3px',
+    marginRight: theme.spacing(1)
   },
-  header: ({ expanded }) => ({
-    display: 'flex',
-    flexDirection: 'row',
-    alignItems: 'flex-start',
-    justifyContent: 'space-between',
-    width: '100%',
-    ...(expanded ? { minHeight: theme.spacing(3) } : { height: theme.spacing(3) }),
-    padding: 0
+  '&:hover': {
+    background: 'none'
+  }
+}));
+
+const HideButton = styled(ExpandLess)(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  width: theme.spacing(3),
+  height: theme.spacing(3),
+  color: theme.palette.text.secondary
+}));
+
+const ShowButton = styled(ExpandMore)(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  width: theme.spacing(3),
+  height: theme.spacing(3),
+  color: theme.palette.text.secondary
+}));
```

Comment:
As these styles are identical, please consider adding it to the parent `<Icon> `instead, to avoid repeating the same code.

Reply:
You are right!, I've readdressed duplicated code